qiskit/transpiler/passes/basis/decompose.py

```diff
@@ -123,9 +133,20 @@ def _should_decompose(self, node: DAGOpNode) -> bool:
 
 
 def _node_to_dag(node: DAGOpNode) -> DAGCircuit:
+    # Control flow is already handled separately, however that does not capture
+    # c_if, which we are treating here. We explicitly ignore the condition attribute,
+    # which will be handled by ``substitute_node_with_dag``, so we create a copy of the node
+    # and set the condition to None. Once ``c_if`` is remoevd for 2.0, this block can go, too.
```

Comment:
Nitpick: would it make sense to make this a TODO? (I personally search for TODOs explicitly so it's easier for me to then find these comments for removal)
```suggestion
    # and set the condition to None. Once ``c_if`` is removed for 2.0, this block can go, too.
```